packages/cli/commands/project/watch.js

```diff
@@ -76,13 +81,36 @@ exports.handler = async options => {
 
   validateProjectConfig(projectConfig, projectDir);
 
-  await createWatcher(
+  await ensureProjectExists(accountId, projectConfig.name);
+
+  const { results } = await fetchProjectBuilds(
     accountId,
-    projectConfig,
-    projectDir,
-    handleBuildStatus,
-    handleSigInt
+    projectConfig.name,
+    options
   );
+
+  const startWatching = async () => {
+    await createWatcher(
+      accountId,
+      projectConfig,
+      projectDir,
+      handleBuildStatus,
+      handleSigInt
+    );
+  };
+
+  // Upload all files if no build exists for this project yet
+  if (!results || !results.length) {
+    logger.log(i18n(`${i18nKey}.logs.initialUploadRequired`));
+    await handleProjectUpload(
+      accountId,
+      projectConfig,
+      projectDir,
+      startWatching
+    );
```

Comment:
Should we also handle build & deploy status here as well?

Reply:
Yeah I wasn't sure about that. My thinking for omitting it was to get to the "watching for file changes" output quicker. The watch command doesn't necessarily need the initial build to be completed before starting up. It just needs at least one build to exists. That being said, I don't feel too strongly about it. Maybe we can loop @markhazlewood in for some design input.

Mark the **TLDR** is:
This PR adds new functionality to the `hs project watch` command. Now it will detect that a project has never been uploaded to HubSpot, and will automatically upload all of the project's files before starting up the "watch" functionality. I'm re-using the same logic that we use for the `hs project upload` command, but I left out the part where the CLI shows you a visual of the components building and deploying (I explain my thinking for doing that above). Do you have any thoughts on it?

Reply:
@anthmatic are you ok with me getting this in without the initial upload build/deploy logging? Should be easy enough to add later on if we decide that we want it. I just want to move this along to fill the gap.

I think I may do a follow up PR to move around some of the project util files anyways. We have some naming conflicts going on in the repo where we use the word "project" for two different things (github repos and also developer projects).

Reply:
Here's how it currently looks:
![Feb-09-2022 16-22-24](https://user-images.githubusercontent.com/6654014/153292258-c6ba3571-10a3-47ff-bc8c-6c16047c9bc3.gif)

Reply:
Yep! This looks good otherwise!